File: Packets/src/main/java/me/ImSpooks/core/packets/security/shared/SharedEncryption.java
package me.ImSpooks.core.packets.security.shared;

import me.ImSpooks.core.packets.security.SecurityEncryption;

/**
 * Created by Nick on 02 okt. 2019.
 * Copyright © ImSpooks
 */
public class SharedEncryption {
	private static SecurityEncryption encryption;

	public static void setEncryption(SecurityEncryption e) {
		if (e == null)
			throw new IllegalArgumentException("Encryption may not be null");
		encryption = e;
	}

	public static SecurityEncryption getEncryption() {
		return encryption;
	}
}

File: ClientBridge/src/main/php/client/packets/security/shared/SharedEncryption.php
<?php
/**
 * Created by Nick on 21 okt. 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\security\shared;


use client\packets\security\SecurityEncryption;
use InvalidArgumentException;

class SharedEncryption {

    //TODO convert to 7.4
    /** @var SecurityEncryption */ private static $encryption;

	public static function setEncryption(SecurityEncryption $encryption) {
		if ($encryption == null)
			throw new InvalidArgumentException("Encryption may not be null");
		self::$encryption = $encryption;
	}

    public static function getEncryption(): SecurityEncryption {
		return self::$encryption;
	}

}